### PR TITLE
Azure Container Apps Deployment Improvements

### PR DESCRIPTION
**Pull request by black-bryan**
Overall improvements to the Azure Container App deployment guide.

* Adding support for deploying the 1Password SCIM Bridge to Azure Container apps using a YAML file. 
  * Prioritizing the `yaml` deployment over the `az containerapp create` commands for separate containers. 
  * Keeping the commands available in the guide for anybody who wants to still created based off the commands.
* Shortening environment variable names to simply and shorten commands used during deployment
* Clarity on some administrative steps.
* Adding `--enable-workload-profiles false` to the `az containerapp env create` command to accommodate the [change made by Azure on December 5th version 2.55.0](https://learn.microsoft.com/en-us/cli/azure/release-notes-azure-cli#containerapp-1) in the AZ CLI tooling.

### CODE REVIEW

File: beta/azure-container-apps/ReadMe.md [hunk 1]
```diff
@@ -60,95 +55,129 @@ Deploying 1Password SCIM Bridge on Azure Container Apps comes with a few benefit
...
+
+    > **Note:** This command can be skipped if you already know the region you want to deploy to and if that region supports Container Apps. 
+
 4. Define variables for the deployment using the following example in the Cloud Shell, _(using the bash or PowerShell syntax for the commands)_.
 
-    > **Note**
-    >
-    >The ContainerAppName can contain lowercase letters, numerals, and hyphens. It must be between 2 and 32 characters long, and cannot start or end with a hyphen.
-    > The location utilizes the name field of the az account list-locations command. 
+    > **Note:** The ConAppName can contain lowercase letters, numerals, and hyphens. It must be between 2 and 32 characters long, and cannot start or end with a hyphen.
```
[the remark this answers is not in the document]
Comment (black-bryan) [reply beta/azure-container-apps/ReadMe.md:68]: Tested during a call today, and found it also does not support starting with a number for the Container App Name
```suggestion
    > **Note:** The ConAppName variable (Container App Name) can contain lowercase letters, numerals, and hyphens. It must be between 2 and 32 characters long, and cannot start or end with a hyphen or cannot start with a number.
```

File: beta/azure-container-apps/ReadMe.md [hunk 2]
```diff
@@ -60,95 +55,129 @@ Deploying 1Password SCIM Bridge on Azure Container Apps comes with a few benefit
...
+        ```pwsh
+        az containerapp create --resource-group $ResourceGroup --environment $ConAppEnv --name $ConAppName  --secrets scimsession="$([Convert]::ToBase64String([IO.File]::ReadAllBytes((Join-Path /home/$Env:USER/ 'scimsession'))) )"
+        ```
+
+9. Deploy your SCIM bridge containers based off the template file:
+
+    Upload the [aca-op-scim-bridge.yaml](aca-op-scim-bridge.yaml) file to the Cloud Shell:
+    - In a separate browser window, download the [aca-op-scim-bridge.yaml](aca-op-scim-bridge.yaml) (selecting the download icon from the top right) from our GitHub repository. 
+    - Click the “Upload/Download files” button and choose Upload.
+    - Find the `aca.op-scim-bridge.yaml` file that you saved to your computer and choose it.
```
Comment (plttn) [inline beta/azure-container-apps/ReadMe.md:130]: We could potentially have them curl the yaml down, save an upload/download step?

File: beta/azure-container-apps/ReadMe.md [hunk 3]
```diff
@@ -60,95 +55,129 @@ Deploying 1Password SCIM Bridge on Azure Container Apps comes with a few benefit
...
-        --environment $ContainerAppEnvironment \
-        --ingress external --target-port 3002 \
-        --cpu 0.25 --memory 0.5Gi \
-        --min-replicas 1 --max-replicas 1 \
-        --secrets scimsession="$(cat /home/$USER/scimsession | base64)" \
-        --env-vars OP_REDIS_URL="redis://localhost:6379" OP_SESSION=secretref:scimsession
+    az containerapp update --resource-group $ResourceGroup --name $ConAppName --yaml aca-op-scim-bridge.yaml --query properties.configuration.ingress.fqdn
+    ```
+
+10. Open the domain name listed in a separate browser tab to test your connection. You can sign in to your SCIM bridge using your bearer token.
```
Comment (plttn) [inline beta/azure-container-apps/ReadMe.md:137]: I've ran into this enough times, can we specify "https://foobar" here? I've had to tell folks "go ahead and put https:// ahead of it"

**Review by plttn — CHANGES_REQUESTED** [review]
Again, minor changes.

**Review by plttn — APPROVED** [review]
Looks good to me, I haven't had a chance to do a test deployment from Windows yet however, so I'll count this as a half-approval.

**Comment by black-bryan** [comment]
@scottisloud, my guess is the fact that the `curl` command or `Invoke-RestMethod -Uri` command are now pointing to the path where the .yaml will live when this is merged caused you the issue. Sorry I should have called this out as part of the review notes.
As it does not look like there are any "real" containers created in your Container App. You can test with `curl https://raw.githubusercontent.com/1Password/scim-examples/solutions/bb/aca-yaml/beta/azure-container-apps/aca-op-scim-bridge.yaml --output aca-op-scim-bridge.yaml --silent` for and repeat the other command in step #9.

**Pull request merged**